internal/extensionserver/mcproute.go

```diff
@@ -304,6 +326,13 @@ func (s *Server) extractMCPBackendFiltersFromMCPProxyListener(listeners []*liste
 				continue // Skip chains without HCM.
 			}
 
+			// All listeners will have the same access log configuration, as they all belong to the same gateway
+			// and share the infrastructure. we can just return the first not-empty access log config and use that
+			// to configure the MCP backend listener with the same settings.
+			if len(accessLogConfig) == 0 {
+				accessLogConfig = httpConManager.AccessLog
+			}
```

Comment:
i would drop this if block for simplicity but it's alight (this makes me think this if block makes the first slice special but reality is any slice from chain should be fine)

Reply:
yeah I like reassign without the if-block better but it's ok to keep it as-is

Reply:
sg. I'll change